test/reg-test/fam-api-reg/fam_control_path_reg_test.cpp

```diff
@@ -0,0 +1,209 @@
+
+#include <fam/fam.h>
+
+#include "cis/fam_cis_client.h"
+#include "common/fam_test_config.h"
+#define BIG_REGION_SIZE 214748364
+#define BLOCK_SIZE 1048576
+#define RESIZE_REGION_SIZE 1048576
+#define NAME_BUFF_SIZE 255
+#define DATA_ITEM_SIZE 1048576
+
+
+//#define MESSAGE_SIZE 12
+
+using namespace std;
+using namespace openfam;
+
+int *myPE;
+fam *my_fam;
+Fam_CIS_Client *cis;
+Fam_Options fam_opts;
+int NUM_MM_ITERATIONS = 100;
+
+
+// Test case 1 - fam_copy and fam_copy_wait test (success).
+TEST(FamDataPath, CreateDestroyRegion) {
+   Fam_Region_Descriptor **descLocal =
+   new Fam_Region_Descriptor *[NUM_MM_ITERATIONS];
+   const char *testRegionLocal = get_uniq_str("testLocal", my_fam);
+
+   for (int i = 0; i < NUM_MM_ITERATIONS; i++) {
+       char regionInfo[NAME_BUFF_SIZE];
+       sprintf(regionInfo, "%s_%d", testRegionLocal, i);
+       EXPECT_NO_THROW(descLocal[i] = my_fam->fam_create_region(
+                            regionInfo, BLOCK_SIZE, 0777, RAID1));
+       EXPECT_NE((void *)NULL, descLocal[i]);
+   }
+   EXPECT_NO_THROW(my_fam->fam_barrier_all());
+   for (int i = 0; i < NUM_MM_ITERATIONS; i++) {
+        EXPECT_NO_THROW(my_fam->fam_destroy_region(descLocal[i]));
+   }
+   EXPECT_NO_THROW(my_fam->fam_barrier_all());
+   free((void *)testRegionLocal);
+}
+
+TEST(FamDataPath, FamAllocateDellocateMultiple1) {
+    Fam_Descriptor **itemLocal = new Fam_Descriptor *[NUM_MM_ITERATIONS];
+    Fam_Region_Descriptor *descLocal;
+    const char *firstItemLocal = get_uniq_str("firstLocal", my_fam);
+    const char *testRegionLocal = get_uniq_str("testLocal", my_fam);
+
+    EXPECT_NO_THROW(descLocal = my_fam->fam_create_region(
+                        testRegionLocal, BIG_REGION_SIZE, 0777, RAID1));
+    EXPECT_NE((void *)NULL, descLocal);
+
+    for (int i = 0; i < NUM_MM_ITERATIONS; i++) {
+        char itemInfo[NAME_BUFF_SIZE];
+        sprintf(itemInfo, "%s_%d", firstItemLocal, i);
+        // Allocating data items
+        EXPECT_NO_THROW(itemLocal[i] = my_fam->fam_allocate(
+                            itemInfo, DATA_ITEM_SIZE, 0777, descLocal));
+        EXPECT_NE((void *)NULL, itemLocal[i]);
+    }
+    EXPECT_NO_THROW(my_fam->fam_barrier_all());
+    for (int i = 0; i < NUM_MM_ITERATIONS; i++) {
+        EXPECT_NO_THROW(my_fam->fam_deallocate(itemLocal[i]));
+    }
+    EXPECT_NO_THROW(my_fam->fam_barrier_all());
+    EXPECT_NO_THROW(my_fam->fam_destroy_region(descLocal));
+    free((void *)testRegionLocal);
+    free((void *)firstItemLocal);
+}
+
+TEST(FamDataPath, FamResizeRegion) {
+ Fam_Region_Descriptor *descLocal;
+ const char *testRegionLocal = get_uniq_str("testLocal", my_fam);
+ EXPECT_NO_THROW(descLocal = my_fam->fam_create_region(
+                     testRegionLocal, RESIZE_REGION_SIZE, 0777, RAID1));
+ EXPECT_NE((void *)NULL, descLocal);
+
+ EXPECT_NO_THROW(my_fam->fam_barrier_all());
+ Fam_Stat *info = (Fam_Stat *)malloc(sizeof(Fam_Stat));
+ for (int i = 0; i < NUM_MM_ITERATIONS; i++) {
```

Comment:
I think we need to rethink this resize in a loop logic - We do not need to resize NUM_MM_ITERATIONS times. Couple of iterations should be enough